client/js/test/table-model.test.js

```diff
@@ -15,4 +15,23 @@ describe('table-model', () => {
     // inspect the resulting state
     expect(model.getValue(location)).toBe('foo');
   });
+
+  it('can add columns, store sums and return sums', () => {
```

Comment:
Make a separate test for each functionality you want to test. Testing multiple functionalities in one test leads to imperfect and confusing tests; it's much harder to debug.

Additionally, your explanation ('can add columns, store sums') must describe in detail what functionality you are testing. "it can add columns" --- when? At all? When there are only numbers as inputs? When there are non-number inputs only? When there is a mix of number and non-number inputs?

Add several more tests to examine edge cases.